File: features/snippets/external_storage/s3_setup/s3_driver_create.py
import os

import aioboto3  # type: ignore[import-not-found]
from temporalio.contrib.aws.s3driver import S3StorageDriver
from temporalio.contrib.aws.s3driver.aioboto3 import new_aioboto3_client

AWS_PROFILE = os.environ.get("AWS_PROFILE")
AWS_REGION = os.environ.get("AWS_REGION", "us-east-2")


async def create_s3_driver():
    # @@@SNIPSTART python-s3-driver-create
    session = aioboto3.Session(profile_name=AWS_PROFILE, region_name=AWS_REGION)
    async with session.client("s3") as s3_client:
        driver = S3StorageDriver(
            client=new_aioboto3_client(s3_client),
            bucket="my-temporal-payloads",
        )
        # @@@SNIPEND
        return driver

File: features/snippets/external_storage/s3_setup/s3_driver_create.ts
import { S3Client } from '@aws-sdk/client-s3';
import { S3StorageDriver } from '@temporalio/external-storage-s3';
import { AwsSdkS3StorageDriverClient } from '@temporalio/external-storage-s3-aws-sdk';
/* eslint-disable @typescript-eslint/no-unused-vars */

{
  // @@@SNIPSTART typescript-s3-driver-create
  const s3Client = new S3Client({ region: 'us-east-2' });

  const driver = new S3StorageDriver({
    client: new AwsSdkS3StorageDriverClient(s3Client),
    bucket: 'my-temporal-payloads',
  });
  // @@@SNIPEND
}
